src/test_servers.py

```diff
@@ -2,33 +2,164 @@
 '''These are tests for client.py and server.py'''
 import pytest
 
-from server import response_ok, response_error, response_deconstructor
+from server import response_ok, response_deconstructor,\
+ request_deconstructor, parse_req, response_decision
+
+from client import client
 
 RESPONSE_TABLE = [
    ('content-length', response_deconstructor(response_ok())[3]),
    ('content-type', response_deconstructor(response_ok())[3]),
    ('date', response_deconstructor(response_ok())[3]),
 ]
 
-FIRST_LINE_TABLE = [
+FIRST_LINE_BODY_LENGTH_TABLE = [
    ('HTTP/1.1', response_deconstructor(response_ok())[0]),
    ('200', response_deconstructor(response_ok())[1]),
    ('OK', response_deconstructor(response_ok())[2]),
    ('8', response_deconstructor(response_ok())[4]),
-   ('2', response_deconstructor(response_ok())[5]),
-   ('HTTP/1.1', response_deconstructor(response_error())[0]),
-   ('500', response_deconstructor(response_error())[1]),
-   ('Internal_Server_Error', response_deconstructor(response_error())[2]),
-   ('7', response_deconstructor(response_error())[4]),
-   ('2', response_deconstructor(response_error())[5])
```

Comment:
In the lines above, I count 7 calls to `response_deconstructor(response_ok))`.  Can you make only 1 call and then use the result in 7 places?